h/services/group_create.py

```diff
@@ -38,7 +34,7 @@ def create_private_group(self, name, userid, **kwargs):
         return self._create(
             name=name,
             userid=userid,
-            type_flags=PRIVATE_GROUP_TYPE_FLAGS,
```

Comment:
**This is not a suggestion**, just my playing with "how would this look as an Enum".

I kinda prefer enum over dict if the keys are defined beforehand, but again, I was just playing with it:

```diff
diff --git a/h/models/group.py b/h/models/group.py
index edfa6b717..874c53630 100644
--- a/h/models/group.py
+++ b/h/models/group.py
@@ -187,11 +187,10 @@ class Group(Base, mixins.Timestamps):
         return slugify.slugify(self.name)
 
     @property
-    def type(self):
+    def type(self) -> str:
         """
         Get the "type" of this group, e.g. "open" or "private".
 
-        :rtype: string
         :raises ValueError: if the type of the group isn't recognized
 
         """
@@ -201,9 +200,9 @@ class Group(Base, mixins.Timestamps):
             writeable_by=self.writeable_by,
         )
 
-        for type_, type_flags in GROUP_TYPE_FLAGS.items():
-            if self_type_flags == type_flags:
-                return type_
+        for type_ in GROUP_TYPE_FLAGS:
+            if type_ == self_type_flags:
+                return type_.name
 
         raise ValueError(
             "This group doesn't seem to match any known type of group. "
@@ -211,9 +210,9 @@ class Group(Base, mixins.Timestamps):
         )
 
     @type.setter
-    def type(self, value):
+    def type(self, value: str):
         try:
-            new_type_flags = GROUP_TYPE_FLAGS[value]
+            new_type_flags = GROUP_TYPE_FLAGS[value].value
         except KeyError as err:
             raise ValueError() from err
 
@@ -235,18 +234,20 @@ class Group(Base, mixins.Timestamps):
 
 TypeFlags = namedtuple("TypeFlags", "joinable_by readable_by writeable_by")
 
-GROUP_TYPE_FLAGS = {
-    "open": TypeFlags(
+from enum import Enum
+
+
+class GROUP_TYPE_FLAGS(TypeFlags, Enum):
+    open = TypeFlags(
         joinable_by=None,
         readable_by=ReadableBy.world,
         writeable_by=WriteableBy.authority,
-    ),
-    "private": TypeFlags(
+    )
+    private = TypeFlags(
         joinable_by=JoinableBy.authority,
         readable_by=ReadableBy.members,
         writeable_by=WriteableBy.members,
-    ),
-    "restricted": TypeFlags(
+    )
+    restricted = TypeFlags(
         joinable_by=None, readable_by=ReadableBy.world, writeable_by=WriteableBy.members
-    ),
-}
+    )
```